server/services/IndexService.ts

```diff
@@ -121,30 +101,25 @@ export default class IndexService {
     }
   };
 
-  // given a list of indexUuids return the managed status of each (true, false, N/A)
-  _getManagedStatus = async (request: KibanaRequest, indexUuids: string[]): Promise<{ [indexUuid: string]: string }> => {
+  _getManagedStatus = async (request: KibanaRequest, indexNames: string[]): Promise<{ [indexName: string]: string }> => {
     try {
-      const searchParams: RequestParams.Search = {
-        index: INDEX.OPENDISTRO_ISM_CONFIG,
-        size: indexUuids.length,
-        body: { _source: "_id", query: { ids: { values: indexUuids } } },
-      };
-      const { callAsCurrentUser: searchCallWithRequest } = this.esDriver.asScoped(request);
-      const results: SearchResponse<any> = await searchCallWithRequest("search", searchParams);
-      const managed: { [indexUuid: string]: string } = results.hits.hits.reduce(
-        (accu: object, hit: { _id: string }) => ({ ...accu, [hit._id]: "Yes" }),
-        {}
-      );
-      return indexUuids.reduce((accu: object, value: string) => ({ ...accu, [value]: managed[value] || "No" }), {});
-    } catch (err) {
-      // If the config index does not exist then nothing is being managed
-      if (err.statusCode === 404 && err.body.error.type === "index_not_found_exception") {
-        return indexUuids.reduce((accu, value) => ({ ...accu, [value]: "No" }), {});
+      const explainParamas = { index: indexNames.toString() };
+      const { callAsCurrentUser: callWithRequest } = this.esDriver.asScoped(request);
+      const explainResponse: ExplainResponse = await callWithRequest("ism.explain", explainParamas);
+
+      const managed: { [indexName: string]: string } = {};
+      for (const indexName in explainResponse) {
+        if (indexName === "total_managed_indices") continue;
+        const explain = explainResponse[indexName] as ExplainAPIManagedIndexMetaData;
+        managed[indexName] = explain["index.opendistro.index_state_management.policy_id"] === null ? "No" : "Yes";
```

Comment:
Is this still accurate? We removed usage of policy_id setting so don't think we can use this anymore.

Reply:
This is still used, retrieved the policyID from managed-index and show here.
We don't have to breaking change this;
and this is still a useful indicator of whether the index is managed or not, it shows null if this index is not managed